packages/utils/src/common/to_async_iterator.ts

```diff
@@ -23,8 +28,18 @@ export async function toAsyncIterator<T extends IDecodedMessage>(
     options
   );
 
+  const isWithTimeout = Number.isInteger(iteratorOptions?.timeoutMs);
+  const timeoutMs = iteratorOptions?.timeoutMs ?? 0;
+  const startTime = Date.now();
+
   async function* iterator(): AsyncIterator<T> {
     while (true) {
+      if (isWithTimeout && Date.now() - startTime >= timeoutMs) {
+        return;
+      }
+
+      await wait(60);
```

Comment:
This is required to make generator idle and unblock main thread.

Reply:
Shall we add that as a code comment?

Reply:
Will add a js-doc to the function.